### DISCUSSION TITLE
Extract/infer a pydantic model from a prexisting database

### DISCUSSION THREAD

**Discussion by mhlr** [fastapi/sqlmodel#592] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
NA
```

### Description

* Have a prexisting, possibly complex, sql database instance
* No prexisting python model classes
* Automatically generate the pydantic model corresponding to the tables in the database.

### Operating System

Linux

### Operating System Details

Ubuntu-22.04

### SQLModel Version

0.0.8

### Python Version

3.9

### Additional Context

(How) can I use SQLModel to generate a pydantic model for an existing SQL database, say MS SQL Server for concreteness (but any database really)?

**Comment by YuriiMotov** [comment]
There is no such tool for now, but this feature in [SQLModel's roadmap](https://github.com/fastapi/sqlmodel/issues/654):
<img width="712" height="348" alt="image" src="https://github.com/user-attachments/assets/8bd4a756-3a07-4b29-b94a-a56872cd3299" />
